apps/builder/middleware.js

```diff
@@ -428,31 +428,40 @@ if (process.env.NODE_ENV !== "production") {
     });
 
     passthrough = async (ctx) => {
-      const devUrl = new URL(ctx.req.url);
-      devUrl.port = "3000";
-
-      const raw = ctx.req.raw;
-
-      const req = new Request(devUrl, {
-        headers: raw.headers,
-        method: raw.method,
-        body: raw.body,
-        redirect: "manual",
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-ignore
-        duplex: "half",
-      });
-
-      const res = await fetch(req);
-
-      const responseHeaders = new Headers(res.headers);
-      responseHeaders.delete("content-encoding");
-
-      return new Response(res.body, {
-        status: res.status,
-        statusText: res.statusText,
-        headers: responseHeaders,
-      });
+      try {
+        const devUrl = new URL(ctx.req.url);
+        devUrl.port = "3000";
```

Comment:
not new, but shoudn't be hardcoded

Reply:
This is dev only mode. I don't see where we set remix port to reuse here.

Reply:
its the default used in remix I think, we should probably pass it from our side to have access to it